r_sort.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+from merge_sort import timed_func
+
+
+@timed_func
+def r_sort(sequence):
+    digit = 1
+    while True:
+        bins = [[] for i in range(10)]
+        for x in sequence:
+            # the expression in bins returns the number in digit place
+            bins[x % 10 ** digit // 10 ** (digit -1)].append(x)
+        sequence = [number for x in bins for number in x]
+        digit += 1
+        if len(sequence) == len(bins[0]):
+            return sequence
+
+
+def str_index(_str, digit):
+    """Returns the ord of the text at the index of digit"""
+    try:
+        index = ord(_str[digit])
```

Comment:
This naming is a bit confusing.  `index` usually means "position in a list" but here `digit` is used for that and `index` is used to mean 'the integer value I will use to represent that letter'.  I know that this value will be *used* as an index outside of this function, but here the naming is confusing.